### PR TITLE
Preserving mast docs vertical real estate 

### PR DESCRIPTION
**Pull request by mstcyr2**
Removing redundancies and excess while preserving the flow of the document.

### CODE REVIEW

File: docs/mast/mast.rst [hunk 1]
```diff
@@ -636,13 +551,7 @@ the user that they might be getting a subset of the true result set.
...
      8150896 0.006357935819940561 322.49334740450234 ...          nan          0
    100906349  0.00808206428937523  322.4932839715549 ...          nan          0
-   105434103 0.011947078376104195 322.49324000530777 ...          nan          0
-   103116183  0.01274757103013683  322.4934207202404 ...          nan          0
-    45593349 0.013026569623011767  322.4933878707698 ...          nan          0
-   103700905  0.01306760650244682  322.4932769229944 ...          nan          0
-   102470085 0.014611879195009472 322.49311034430366 ...          nan          0
-    93722307  0.01476438046135455 322.49348351134466 ...          nan          0
-    24781941 0.015234351867433582 322.49300148743345 ...          nan          0
+   ...
```
Comment (bsipocz) [inline docs/mast/mast.rst:554]: e.g. this example explicitly prints out 10 lines, so I would keep all those 10 lines. 

The other changes where there is a longer ellipse, sure, culling those back sounds fine.

**Comment by bsipocz** [comment]
@mstcyr2 @jaymedina - At this point it may make sense to chop up the mast documentation into several pages, both into separate rst files that are lined from mast.rst. That would in fact help with testing the documentation, too (see https://github.com/astropy/astroquery/issues/2505)

File: docs/mast/mast.rst (hunk 1 above)
Comment (mstcyr2) [reply docs/mast/mast.rst:554]: Would it be okay to change the example to explicitly print 3 lines instead? A lot of the examples I shortened also use 10.

Comment (bsipocz) [reply docs/mast/mast.rst:554]: @jaymedina should comment on that. 
I feel the 10 is more reasonable for a use case, but I don't really mind either case, just don't use ellipses for these short examples.

**Review by jaymedina — CHANGES_REQUESTED** [review]
This is looking good to me, I just had one small suggestion. I've sent this PR to Dick Shaw who would also like to give input on it, and I'll relay his comments later this week in a separate review.

**Comment by jaymedina** [comment]
Dick Shaw has some comments that are slightly out of scope of this PR so we've agreed to tackle them in a separate one, and this one's good to go. I'll be wrapping up this PR for Makayla while she's on break, let me know if there's anything that needs to be updated before this is ready for merge @bsipocz. Thanks!

**Review by bsipocz — APPROVED** [review]
Rendering looks OK, and I defer the content review to the MAST colleagues, so this is indeed good to go.

Thanks @mstcyr2!

**Comment by bsipocz** [comment]
> so this is indeed good to go.

except that I see failing tests. Nevertheless, I would go ahead and merge this, but would like to ask to follow-up on the failures (they are already reported as part of https://github.com/astropy/astroquery/issues/2801). Thanks to this PR, now it'll be more efficient to fix the issues as one doesn't need to rerun all of the docs over and over again.

**Comment by bsipocz** [comment]
FYI:

```
collected 6 items                                                                                                      

docs/mast/mast.rst .                                                                                             [ 16%]
docs/mast/mast_catalog.rst F                                                                                     [ 33%]
docs/mast/mast_cut.rst .                                                                                         [ 50%]
docs/mast/mast_mastquery.rst .                                                                                   [ 66%]
docs/mast/mast_missions.rst .                                                                                    [ 83%]
docs/mast/mast_obsquery.rst F                                                                                    [100%]

======================================================= FAILURES =======================================================
______________________________________________ [doctest] mast_catalog.rst ______________________________________________
254 
255 Given an HSC Match ID, return all catalog results.
256 
257 .. doctest-remote-data::
258 
259    >>> from astroquery.mast import Catalogs
260    ...
261    >>> catalog_data = Catalogs.query_object("M10", radius=.02, catalog="HSC")
262    >>> matchid = catalog_data[0]["MatchID"]
263    >>> print(matchid)
Expected:
    63980492
Got:
    7542452

/Users/bsipocz/munka/devel/astroquery/docs/mast/mast_catalog.rst:263: DocTestFailure
_____________________________________________ [doctest] mast_obsquery.rst ______________________________________________
049       science    SPITZER_SHA    SSC Pipeline ...    nan 19000016510      0.0
050       science    SPITZER_SHA    SSC Pipeline ...    nan 19000016510      0.0
051       science    SPITZER_SHA    SSC Pipeline ...    nan 19000016510      0.0
052 
053 Optional parameters must be labeled. For example the query above will produce
054 an error if the "radius" field is not specified.
055 
056 .. doctest-remote-data::
057 
058    >>> obs_table = Observations.query_object("M8", ".02 deg")
Differences (unified diff with -expected +actual):
    @@ -1,3 +1,12 @@
     Traceback (most recent call last):
    -...
    -TypeError: ObservationsClass.query_object_async() takes 2 positional arguments but 3 were given
    +  File "/Users/bsipocz/.pyenv/versions/3.9.1/lib/python3.9/doctest.py", line 1336, in __run
    +    exec(compile(example.source, filename, "single",
    +  File "<doctest mast_obsquery.rst[6]>", line 1, in <module>
    +    obs_table = Observations.query_object("M8", ".02 deg")
    +  File "/Users/bsipocz/munka/devel/astroquery/astroquery/utils/class_or_instance.py", line 25, in f
    +    return self.fn(obj, *args, **kwds)
    +  File "/Users/bsipocz/munka/devel/astroquery/astroquery/utils/process_asyncs.py", line 26, in newmethod
    +    response = getattr(self, async_method_name)(*args, **kwargs)
    +  File "/Users/bsipocz/munka/devel/astroquery/astroquery/utils/class_or_instance.py", line 25, in f
    +    return self.fn(obj, *args, **kwds)
    +TypeError: query_object_async() takes 2 positional arguments but 3 were given

/Users/bsipocz/munka/devel/astroquery/docs/mast/mast_obsquery.rst:58: DocTestFailure
=============================================== short test summary info ================================================
FAILED docs/mast/mast_catalog.rst::mast_catalog.rst
FAILED docs/mast/mast_obsquery.rst::mast_obsquery.rst
======================================= 2 failed, 4 passed in 642.32s (0:10:42) ========================================
```

**Pull request merged**